lib/api/model.js

```diff
@@ -15,6 +17,101 @@ function get (modelName, passByReference) {
     return clone(linz.get('models')[modelName]);
 }
 
+/**
+ * Get permission data for a model
+ * @param {Object}   req      Request data
+ * @param {String}   modelName Model name
+ * @param {Function} cb        Callback function
+ * @return {Object}
+ */
+function getPermissions(req, modelName, cb) {
+
+    if (!linz.get('disabled permission cache') && !req.linz.cache.permissions.invalidate && req.session.permissions && req.session.permissions[modelName]) {
+        return cb(null, req.session.permissions[modelName]);
+    }
+
+    var model = get(modelName);
```

Comment:
I think you could safely retrieve a reference, rather than a copy here. We're not altering the model object so it should be fine, and no need for duplication...